### DISCUSSION TITLE
Getting the previous version of a trigger/pipeline

### DISCUSSION THREAD

**Discussion by MattIPv4** [concourse/concourse#8197] [Help & Support]
Hey folks,

I'm making use of the git resource to trigger a pipeline, and I'd really like to be able to generate a diff as part of my pipeline logic.

To do this though, I need to know not only the current commit hash the pipeline is running on, but also the last version that it ran on. Concourse definitely has this data somewhere, but I'm struggling with figuring out how to retrieve it from within a running pipeline.

Any ideas would be much appreciated.

**Comment by taylorsilva** [comment]
There is sadly no way (that I can think of) to get the current and previous version of a git commit passed into a job.

When concourse starts a build for a job it only passes in the current/pinned versions of all resources. Concourse does "know" what the previous version passed in was, in the sense that this information is in the database. But this information is not actually represented by anything in the code. Concourse doesn't do anything comparisons between versions. 

Now if your goal is to display the commits diff, then you can probably do this in a task by running `git diff HEAD~1` in the git repo passed in by the git resource.

**Reply by MattIPv4** [reply]
Yeah, my intention is to do a commit diff, but we only run this pipeline every so often, so the diff is an unknown number of commits.

Sounds like the simplest solution given Concourse doesn't have the data easily available is probably just to store the last commit hash somewhere that we can reference each run. Ty!
